src/Swift.Bindings/tests/CryptoKit/Foundation.Source.cs

```diff
@@ -0,0 +1,274 @@
+    }
+
+    // <summary>
+    // Represents Swift UnsafeMutablePointer in C#.
+    // </summary>
+    public readonly unsafe struct UnsafeMutablePointer<T> where T : unmanaged
+    {
+        private readonly T* _pointee;
+        public UnsafeMutablePointer(T* pointee)
+        {
+            _pointee = pointee;
+        }
+
+        public T* Pointee => _pointee;
+
+        public static implicit operator T*(UnsafeMutablePointer<T> pointer) => pointer.Pointee;
+
+        public static implicit operator UnsafeMutablePointer<T>(T* pointee) => new(pointee);
+    }
+
+    // <summary>
+    // Represents Swift UnsafeRawPointer in C#.
+    // </summary>
+    public readonly unsafe struct UnsafeRawPointer
+    {
+        private readonly void* _pointee;
+        public UnsafeRawPointer(void* pointee)
+        {
+            _pointee = pointee;
+        }
+
+        public void* Pointee => _pointee;
+
+        public static implicit operator void*(UnsafeRawPointer pointer) => pointer.Pointee;
+
+        public static implicit operator UnsafeRawPointer(void* pointee) => new(pointee);
+    }
+
+    // <summary>
+    // Represents Swift UnsafeMutableRawPointer in C#.
+    // </summary>
+    public readonly unsafe struct UnsafeMutableRawPointer
+    {
+        private readonly void* _pointee;
+        public UnsafeMutableRawPointer(void* pointee)
+        {
+            _pointee = pointee;
+        }
+
+        public void* Pointee => _pointee;
+
+        public static implicit operator void*(UnsafeMutableRawPointer pointer) => pointer.Pointee;
+
+        public static implicit operator UnsafeMutableRawPointer(void* pointee) => new(pointee);
+    }
+
+    // <summary>
+    // Represents Swift UnsafeBufferPointer in C#.
+    // </summary>
+    public readonly unsafe struct UnsafeBufferPointer<T> where T : unmanaged
+    {
+        private readonly T* _baseAddress;
+        private readonly nint _count;
+        public UnsafeBufferPointer(T* baseAddress, nint count)
+        {
+            _baseAddress = baseAddress;
+            _count = count;
+        }
+
+        public T* BaseAddress => _baseAddress;
+        public nint Count => _count;
+    }
+
+    // <summary>
+    // Represents Swift UnsafeMutableBufferPointer in C#.
+    // </summary>
+    public readonly unsafe struct UnsafeMutableBufferPointer<T> where T : unmanaged
+    {
+        private readonly T* _baseAddress;
+        private readonly nint _count;
+        public UnsafeMutableBufferPointer(T* baseAddress, nint count)
+        {
+            _baseAddress = baseAddress;
+            _count = count;
+        }
+
+        public T* BaseAddress => _baseAddress;
+        public nint Count => _count;
+    }
+
+    // <summary>
+    // Represents Swift Foundation.Data in C#.
+    // </summary>
+    [StructLayout(LayoutKind.Sequential, Size = 16)]
+    [InlineArray(16)]
+    public unsafe struct Data : ISwiftObject
+    {
+        private byte _payload;
```

Comment:
Until those its flow https://github.com/dotnet/runtime/pull/108483, this will fail on mono

Reply:
Currently, we test only CoreCLR.